src/components/frame/v5/pages/UserAdvancedPage/partials/FeesForm/hooks.tsx

```diff
@@ -1,64 +1,85 @@
-import { bool, object } from 'yup';
+import { bool, object, InferType } from 'yup';
 import { toast } from 'react-toastify';
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { UseFormReturn } from 'react-hook-form';
 
 import Toast from '~shared/Extensions/Toast';
 import { canUseMetatransactions } from '~utils/checks';
 import { useUpdateUserProfileMutation } from '~gql';
 import { useAppContext } from '~hooks';
+import { notMaybe } from '~utils/arrays';
+import noop from '~utils/noop';
+import { omit } from '~utils/lodash';
 
 export const useFeesForm = () => {
   const metatransactionsValidationSchema = object({
     metatransactionsEnabled: bool<boolean>(),
   }).defined();
 
+  const [formRef, setFormRef] = useState<UseFormReturn<
+    InferType<typeof metatransactionsValidationSchema>
+  > | null>(null);
+
   const metatransactionsAvailable = canUseMetatransactions();
 
-  const { user, updateUser } = useAppContext();
-  const { metatransactionsEnabled, customRpc, decentralizedModeEnabled } =
-    user?.profile?.meta ?? {};
+  const { user, updateUser, userLoading } = useAppContext();
+  const userProfileMeta = user?.profile?.meta;
 
-  const [editUser] = useUpdateUserProfileMutation();
+  const [editUser, { loading: editUserLoading }] =
+    useUpdateUserProfileMutation();
 
   const metatransactionsDefault = metatransactionsAvailable
-    ? metatransactionsEnabled
+    ? userProfileMeta?.metatransactionsEnabled
     : false;
 
-  const handleSubmit = async (values: { metatransactionsEnabled: boolean }) => {
-    await editUser({
-      variables: {
-        input: {
-          id: user?.walletAddress ?? '',
-          meta: {
-            customRpc,
-            decentralizedModeEnabled,
-            ...values,
+  useEffect(() => {
+    if (!formRef) {
+      return noop;
```

Comment:
can't we just `return` why do we need a noop?

Reply:
To fullfill eslint rule https://eslint.org/docs/latest/rules/consistent-return

Reply:
but this rule isn't applicable for breaking out of useEffect

Reply:
maybe we need 
```JSX
return () => {
  unsubscribe();
}
```
here instead. It's probably nagging you because unsubscribe doesn't fulfill the type requirements

Reply:
No, it's because in one place we return `void` and in the other we are returning a function which is treated as inconsistent. I can return `undefined` instead of `noop` if it works for you
![image](https://github.com/JoinColony/colonyCDapp/assets/38806520/faca0a7c-1819-4692-931d-c2bfdae60ebc)
![image](https://github.com/JoinColony/colonyCDapp/assets/38806520/c3b975d2-5603-44bc-b116-99fc25ccd3bf)

Reply:
Changed to `undefined`